numpy_challenge.py

```diff
@@ -0,0 +1,113 @@
+# %%
+import numpy as np
+
+
+def task_1(n):
+    # Solution 1
+
+    arr = np.zeros(n)
+    arr[0] = 0
+    arr[1] = 1
+
+    for i in range(2, n):
+        arr[i] = arr[i - 1] + arr[i - 2]
+
+    print(arr)
+
+    # Solution 2
+
+    arr = np.arange(n)
+    phi = (1 + np.sqrt(5)) / 2
+
+    arr = (phi ** arr - (-1 / phi) ** arr) / np.sqrt(5)
+
+    print(arr)
+
+    # Solution 3
+
+    arr = np.arange(n)
+    phi = (1 + np.sqrt(5)) / 2
+
+    def f(x):
+        return (phi ** x - (-1 / phi) ** x) / np.sqrt(5)
+
+    vecf = np.vectorize(f)
+    arr = vecf(arr)
+    print(arr)
+
+
+def matrix_multiplication(a, b):
+    a = np.asarray(a)
+    b = np.asarray(b)
+
+    return a @ b
+
+
+def multiplication_check(list_mat):
+    list_mat = [np.asarray(a) for a in list_mat]
+
+    m, n = list_mat[0].shape
+
+    for i in range(1, len(list_mat)):
+        if n == list_mat[i].shape[0]:
+            n = list_mat[i].shape[1]
+        else:
+            return False
+
+    return True
```

Comment:
Подход правильный, но можно немного упростить
```suggestion
    list_mat = [np.asarray(a) for a in list_mat]

    for i in range(len(list_mat) - 1):
        if list_mat[i].shape[1] != list_mat[i+1].shape[0]:
            return False
    return True
```